brightnessd.c

```diff
@@ -1032,7 +1060,57 @@ static uint8_t event_loop(struct Tglobalstate *pglobalstate, struct Txcb *pxcb,
 }
 
 ///////////////////////////////////////////////////////////////////////////////
-// parse_uint8_t()
+// spawn_async
+///////////////////////////////////////////////////////////////////////////////
+/** Asynchronously execute a shell command, i.e. don't wait for it to complete.
+
+    @param cmd              the command to be executed in a shell
+*/
+static void spawn_async(char* cmd) {
+    DEBUG("[spawn_async] executing external command: %s\n", cmd);
+    pid_t process = fork();
+    if (process < 0) {
+        ERROR("[spawn_async] fork error");
+    } else if (process == 0) { // child
+        char *argv[64];
```

Comment:
Hm, while `64` is reasonable large, I guess a bounds-check on overflow is necessary here or in `parse_cmd()`, right?

Reply:
Agreed. There is an overflow check now.